src/SensorReading.cpp

```diff
@@ -134,7 +138,7 @@ bool SensorReading::repeated_values(std::deque<float> buffer)
     int first_val = buffer[0];
     for (float val : buffer) {
         if (first_val != val) {
-            return false;
+            return false; //This seems problematic
```

Comment:
Might be misunderstanding it, but if the function is supposed to check for ANY repeated values and not just a duplicate of the first, this should do that:
int first_val;
    for (int i = 0; i < buffer.size(); i++) {
        int first_val = buffer[i];
        for (float val : buffer) {
            if (first_val != val) {
                return false; // Potential solution? from Max
            }
        }
    }

Reply:
You are correct- repeated values should only be called once the value we want to add has been added.

// LJG TODO might need to add whether we want to check for repeated values in the constructor
    // check if value is within expected range
    if (x <= max && x >= min && !repeated_values(buffer)) {
        if (valid) {
            // add value to buffer
            buffer.push_front(x);

            // check if buffer is full
            if (buffer.size() > buffer_size) {
                // remove oldest value from buffer
                buffer.pop_back();
            }
        }
    } else {
        set_invalid();
    }

Should be 

// LJG TODO might need to add whether we want to check for repeated values in the constructor
    // check if value is within expected range
    if (x <= max && x >= min) {
        if (valid) {
            // add value to buffer
            buffer.push_front(x);

            if(repeated_values(buffer)){
                remove value 
                return false
            }

            // check if buffer is full
            if (buffer.size() > buffer_size) {
                // remove oldest value from buffer
                buffer.pop_back();
            }
        }
    } else {
        set_invalid();
    }